avro-codegen/src/main/java/com/linkedin/avroutil1/codegen/SpecificRecordClassGenerator.java

```diff
@@ -499,14 +505,26 @@ private void populateBuilderClassBuilder(TypeSpec.Builder recordBuilder, AvroRec
               "record.$1L = fieldSetFlags()[$2L] ? this.$1L : ($3L) com.linkedin.avroutil1.compatibility.AvroCompatibilityHelper.getSpecificDefaultValue(fields()[$2L])",
               escapedFieldName, fieldIndex, fieldType);
         } else {
-          buildMethodCodeBlockBuilder
-              .beginControlFlow("if ($L != null)", escapedFieldName+"Builder")
+          canThrowMissingFieldException = true;
+          buildMethodCodeBlockBuilder.beginControlFlow("if ($L != null)", escapedFieldName + "Builder")
               .beginControlFlow("try")
               .addStatement("record.$1L = this.$1LBuilder.build()", escapedFieldName)
-              .endControlFlow()
-              .beginControlFlow("catch (org.apache.avro.AvroMissingFieldException  e)")
-              .addStatement("e.addParentField(record.getSchema().getField($S))", escapedFieldName)
-              .addStatement("throw e")
+              .endControlFlow();
+
+
+
+          if (config.getMinimumSupportedAvroVersion().laterThan(AvroVersion.AVRO_1_8)) {
+            buildMethodCodeBlockBuilder.beginControlFlow("catch (org.apache.avro.AvroMissingFieldException e)")
+                .addStatement("com.linkedin.avroutil1.compatibility.exception.AvroUtilMissingFieldException avroUtilException = new com.linkedin.avroutil1.compatibility.exception.AvroUtilMissingFieldException(e)")
```

Comment:
Are these `Exception`s the first cases where our custom codegen will rely on `avro-util` as a dependency?

Reply:
codegen already uses Compatibility helpers, so this will not result in a new dependency. 
Avro uses different exception classes depending on their versions which changes what our generated code looks like